Source/WebKit2/WebProcess/WebCoreSupport/soup/WebFrameNetworkingContext.cpp

```diff
@@ -84,7 +83,7 @@ WebFrameNetworkingContext::WebFrameNetworkingContext(WebFrame* frame)
 NetworkStorageSession& WebFrameNetworkingContext::storageSession() const
 {
     if (frame() && frame()->page()->usesEphemeralSession())
-        return *NetworkStorageSession::storageSession(SessionID::legacyPrivateSessionID());
+        return *NetworkStorageSession::storageSession(frame()->page()->sessionID());
```

Comment:
Right, we should stop using the legacy session ID, but then we also need to remove the ephemeral session check. This has already been fixed upstream in http://trac.webkit.org/changeset/210728 and I think @zdobersek already cherry-picked it.